### ISSUE TITLE
Determine the optimal K 

### ISSUE THREAD

**Issue by Sudolin** [JSB-UCLA/scPNMF#1]
Hi,

Thanks for the great tool.

I am wondering if there is a way to determine the optimal K in the package rather than using the pre-specified one. Also, when batch effect in merged scRNA datasets exists, could we still use the log raw count as the input data? Thanks in advance.

Best,
Leo Lam

**Comment by aileenlikexin** [comment]
Hi Leo, 

Thank you for the comments!

For choosing optimal K, we proposed a metric "dev.ortho" (normalized difference between W^TW and I) in Section S1.1 of our paper, but did not include it here in our R package. The main reason is that calculating dev.ortho requires running PNMF on a sequence of potential K's and thus is time consuming; the performance doesn't differ too much after stability is reached; and we found K=20 reaches stability for most scRNA-seq data empirically. But I can update this K selection part by this weekend if needed.

For the batch effect issue, we recommend the users run scPNMF on each batch separately or try to correct the batch effect first. Otherwise, some features/genes that are not relevant to cell type variations may be selected. We have observed such cases in our previous analysis.

Best,
Kexin
